Users/toml_config.rst

```diff
@@ -0,0 +1,286 @@
+  a coala section.
+
+This is an example of a coala configuration file written in TOML ::
+
+    [cli]
+    bears = 'SpaceConsistencyBear'
+    files = 'src/*.c'
+    use_spaces = true
+
+    [invalidlinks]
+    enabled = false
+    files = [ '**/*.rst', 'README.rst']
+    ignore = 'venv/**'
+    bears = 'InvalidLinkBear'
+
+
+Here tables ``cli`` and ``invalidlinks`` are coala sections.
+The contents of the tables like ``bears``, ``files`` are rules
+that govern a section. To write coala configuration file you will
+be using TOML strings, booleans, integers and arrays as values.
+
+Section Inheritance
+~~~~~~~~~~~~~~~~~~~~
+coala supports section inheritance. You can define section inheritance
+by using the key ``inherits``.
+Extra values can be appended to an inherited setting using the ``appends`` key.
+
+Consider the following configuration file in TOML ::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  [section1]
+  overridable = 3
+  inherits = 'all'
+  appends = 'ignore'
+  ignore = 'vendor2/'
+  other = 'some_value'
+
+  [section2]
+  overridable = 4
+  inherits = 'all'
+  ignore = 'vendor3/'
+  appends = 'ignore'
+  other = 'some_other_value'
+
+
+In the inherited sections above, ``appends`` key specifies that the value of
+``ignore`` in the derived sections must be appended with the value of
+``ignore`` key in the base section.
+
+This is the same file without section inheritance::
+
+  [all]
+  enabled = true
+  overridable = 2
+  ignore = 'vendor1/'
+
+  [section1]
+  enabled = true
+  overridable = 3
+  ignore = ['vendor1/', 'vendor2/']
+  other = 'some_value'
+
+  [section2]
+  enabled = true
+  overridable = 4
+  ignore = ['vendor1/', 'vendor3/']
+  other = 'some_other_value'
+
+
+Consider another example
+
+Config file in TOML
+
+::
+
+ [all]
+ a = 1
+ b = 2
+
+ [java]
+ c = 3
+ d = 4
+
+ [python]
+ p = 5
+ q = 6
+ inherits = [ 'all', 'java']
+
+You can use this syntax to specify multiple inheritance
+The same is coafile appears as
+
+::
+
+ [all]
+ a = 1
+ b = 2
+
+ [java]
+ c = 3
+ d = 4
+
+ [all.python]
+ a = 1
+ b = 2
+ p = 5
+ q = 6
+
+ [java.python]
+ c = 3
+ d = 4
+ p = 5
+ q = 6
+
+Defining Aspects and Tastes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Aspects is an alternative way to configure coala. In this mode, we don't need
+to explicitly state list of bears, coala will choose it automatically based on
+requested aspects in configuration file. To run coala in this mode, we need to
+define `aspects`, `files`, `languages`, and optionally aspect tastes setting.
+See the following example::
+
+  [all]
+  files = '**'
+  aspects = ['aspectname1', 'AspectName2'] # case-insensitive
+  # defining an aspect's taste
+  aspectname1.aspect_taste = 80
+  # we can define subaspect taste through its parent
+  aspectname1.subaspect_taste = ['word1', 'word2', 'word3']
+
+  ['all.python']
+  files = '**.py'
+  language = 'Python'
+  # appending additional aspect
+  appends = 'all'
```

Comment:
If there is a case where `appends` can be `section` name then create a subheading and explain with example (what will happen when you do so)

Reply:
appends can be list If I am correct, mention it here,